src/transforms/kubernetes/mod.rs

```diff
@@ -1 +1,568 @@
+                        break;
+                    }
+                }
+            }
+
+            warn!(
+                message = "Temporarily stoped watching Pod metadata.",
+                reason = %runtime_error
+            );
+
+            error = Some(runtime_error);
+        }
+    }
+
+    /// Extracts metadata from pod and updates metadata map.
+    fn update(&mut self, (pod, event): (Pod, PodEvent)) -> Option<Version> {
+        if let Some(pod_uid) = pod.metadata.as_ref().and_then(|md| md.uid.as_ref()) {
+            let uid: Bytes = pod_uid.as_str().into();
+
+            self.metadata.clear(uid.clone());
+
+            // Insert field values for this pod.
+            for (field, value) in self
+                .fields
+                .iter()
+                .flat_map(|field| field.extract(&pod).unwrap_or_default())
+            {
+                self.metadata
+                    .insert(uid.clone(), Box::new((field, FieldValue(value))));
+            }
+
+            trace!(message = "Pod updated.", %pod_uid);
+
+            if PodEvent::Deleted == event {
+                self.delete_queue
+                    .push_back((uid, Instant::now() + self.cache_ttl));
+            }
+        }
+
+        Version::from_pod(&pod)
+    }
+
+    /// Checks if there are entries to be deleted.
+    fn delete_update(&mut self) {
+        let now = Instant::now();
+        while let Some((pod_uid, deadline)) = self.delete_queue.get(0) {
+            if *deadline <= now {
+                self.metadata.empty(pod_uid.clone());
+
+                trace!(message = "Pod metadata deleted.", pod_uid=?std::str::from_utf8(pod_uid.as_ref()));
+
+                self.delete_queue.pop_front();
+            } else {
+                break;
+            }
+        }
+    }
+}
+
+#[derive(PartialEq, Debug, Clone)]
+struct FieldValue(Value);
+
+// Since we aren't using Eq feature in the evmap, we can impl Eq.
+impl Eq for FieldValue {}
+
+pub struct KubernetesPodMetadata {
+    metadata: ReadHandle<Bytes, Box<(Atom, FieldValue)>>,
+    pod_uid: Atom,
+}
+
+impl Transform for KubernetesPodMetadata {
+    fn transform(&mut self, mut event: Event) -> Option<Event> {
+        let log = event.as_mut_log();
+
+        if let Some(Value::Bytes(pod_uid)) = log.get(&self.pod_uid) {
+            let pod_uid = pod_uid.clone();
+
+            let found = self.metadata.get_and(&pod_uid, |fields| {
+                for pair in fields {
+                    log.insert(pair.0.clone(), (pair.1).0.clone());
+                }
+            });
+
+            if found.is_none() {
+                warn!(
+                    message = "Metadata for pod not yet available.",
+                    pod_uid = ?std::str::from_utf8(pod_uid.as_ref()),
+                    rate_limit_secs = 10
+                );
+            }
+        } else {
+            warn!(
+                message = "Event without field, so it can't be enriched with metadata.",
```

Comment:
We should align these logs with what we have in other transforms:

https://github.com/timberio/vector/blob/master/src/transforms/json_parser.rs#L81

Reply:
Something like this.